flambe/nlp/language_modeling/task.py

```diff
@@ -0,0 +1,259 @@
+        train_batch_size : int, optional
+            The batch size to use during. Default ``32``.
+        val_batch_size : int, optional
+            The batch size to use during evaluation. Default ``32``.
+        train_unroll_size : int, optional
+            The unroll size to use during. Only used if ``oredered``
+            is set. Default ``32``.
+        val_unroll_size : int, optional
+            The unroll size to use during evaluation. Only used if
+            ``oredered`` is set. Default ``32``.
+        embedding_args : Dict[str, Any], optional
+            Keyword arguments to pass the ``Embeddings`` constructor.
+        optimizer : Optional[Optimizer], optional
+            The optimizer to use. Should be provided for training.
+        iter_scheduler : Optional[LRScheduler], optional
+            A learning rate scheduler to call on every training step.
+        eval_scheduler : Optional[LRScheduler], optional
+            A learning rate scheduler to call on every validation step.
+        text_field : Optional[TextField], optional
+            A custom text field to apply to the text inputs.
+        embedder : Optional[Embedder], optional
+            A custom embedder. Overrides ``encoder`` and ``pooling``.
+        model : Optional[Encoder], optional
+            A custom model. Overrides ``encoder`` and ``embedder``.
+
+        See the ``Training`` parent class for other keyword arguments.
+
+        """
+        super().__init__(**kwargs)
+
+        text_field = text_field or TextField()
+
+        # Build vocabularies
+        if build_vocabularies:
+            text, = zip(*dataset)
+            text_field.setup(text)
+
+        self.dataset = dataset
+        self.text_field = text_field
+        self.dataset._set_transforms({'text': self.text_field})
+
+        self.embedding_args = embedding_args or dict()
+
+        self.train_batch_size = train_batch_size
+        self.val_batch_size = val_batch_size
+        self.train_unroll_size = train_unroll_size
+        self.val_unroll_size = val_unroll_size
+
+        self.text_field = text_field
+
+        if encoder is None and embedder is None and model is None:
+            raise ValueError("At least one of encoder, embedder or model must be provided.")
+
+        self.state = None
+        self.encoder = encoder
+        self.embedder = embedder
+        self._model = model
+        self.dropout = dropout
+
+        self.ordered = ordered
+        if tie_weights and mixture_of_softmax > 0:
+            raise ValueError("Cannot tie weights for a mixture of softmax.")
+
+        self.tie_weights = tie_weights
+        self.loss_fn = nn.CrossEntropyLoss(reduction='none')
+        self.metric_fn = BPC() if use_bpc else Perplexity()
+        self.metric_name = 'bpc' if use_bpc else 'perplexity'
+
+        self.optimizer = optimizer
+        self.eval_scheduler = eval_scheduler
+        self.iter_scheduler = iter_scheduler
+
+    def build_model(self):
```

Comment:
There is a lot of typing missing... not only in this file!

Reply:
Will fix